src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,35 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param() params) {
+    try {
+      const user = await this.userService.getUserById(params.id);
+      if (user) {
+        const resp: IResult<unknown> = {
+          status: 'OK',
+          result: user,
+        };
```

Comment:
I dont think this is okay, because this user object will even contain the user's password.

Reply:
Right now User Schema doesnt contain a password field. But i think this is a valid query. We could pass back only certain user fields.